src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -40,67 +43,148 @@ class ContextFilesPanel(
             logger.info(
-                "Opening a file from the used context (projectRelativeFilePath=$projectRelativeFilePath, file=$file)")
-            FileEditorManager.getInstance(project).openFile(file, /*focusEditor=*/ true)
+                "Opening a file from the used context (displayText=${remoteContextFile.displayText})")
+            openInBrowser(project, contextFileFile.uri.path)
           }
         }
-    val goToFile = AnActionLink("@$projectRelativeFilePath", anAction)
+      }
+
+  private fun createLinkPanel(actionText: String, anAction: () -> AnAction): JPanel {
+    val goToFile = AnActionLink(actionText, anAction.invoke())
+    goToFile.toolTipText = actionText
     val panel = JPanel(BorderLayout())
     panel.isOpaque = false
     panel.border = JBUI.Borders.emptyLeft(3)
     panel.add(goToFile, BorderLayout.PAGE_START)
     return panel
   }
 
+  private fun getActionTextForFileLinkAction(
+      contextFileFile: ContextFileFile?,
+      typedContextFile: TypedContextFile
+  ): String {
+    val startLine = contextFileFile?.range?.start?.line
+    val endLine = contextFileFile?.range?.end?.line
+    return buildString {
+      if (typedContextFile is LocalContextFile) {
+        append("@")
+      }
+      append(typedContextFile)
+      if (startLine != null && endLine != null) {
+        if (startLine != endLine) {
+          append(":$startLine-$endLine")
+        }
+      }
+    }
+  }
+
   fun updateContentWith(contextFiles: List<ContextFile>?) {
     if (contextFiles.isNullOrEmpty()) {
       return
     }
 
-    val contextFileNames =
-        contextFiles
-            .map {
-              if (it.repoName != null) {
-                "${project.basePath}/${it.uri.path}"
-              } else {
-                Paths.get(it.uri.withFragment(null).withQuery(null)).absolutePathString()
-              }
+    val groupBy = contextFiles.groupBy { it.repoName != null }
+
+    val localFiles =
+        (groupBy[false] ?: emptyList())
+            .associateBy {
+              val newUri = it.uri.withFragment(null).withQuery(null)
+              LocalContextFile(Paths.get(newUri).absolutePathString())
+            }
+            .toMap()
+
+    val remoteFiles =
+        (groupBy[true] ?: emptyList())
+            .associateBy {
+              val repo = it.repoName.toString().split('/').last()
+              val file = it.uri.path.split("-/").last()
+              RemoteContextFile("$repo $file")
             }
-            .toSet()
+            .toMap()
 
-    ApplicationManager.getApplication().executeOnPooledThread { updateFileList(contextFileNames) }
+    ApplicationManager.getApplication().executeOnPooledThread {
+      updateFileList(localFiles, remoteFiles)
+    }
   }
 
   companion object {
     private val logger = Logger.getInstance(ContextFilesPanel::class.java)
   }
 }
+
+sealed class TypedContextFile(val displayText: String)
+
+class LocalContextFile(displayText: String) : TypedContextFile(displayText)
```

Comment:
Is display text so me kind path or uri?
If so, name should reflect that, `displayTest` sounds a bit like content to me.
I understand it is what we display, and it is maybe not a real uri in any sense, but still...